File: 0377-combination-sum-iv/0377-combination-sum-iv.cpp
class Solution {
public:
    int combinationSum4(vector<int>& nums, int target) {
        std::sort(nums.begin(), nums.end());
        unsigned int buff[1001] = {1, 0};
        for (int i = 1; i <= target; ++i) {
            for (int j: nums) {
                if (i - j < 0) {
                    break;
                }
                buff[i] += buff[i - j];
            }
        }
        return buff[target];
    }
};

File: 0377-combination-sum-iv/0377-combination-sum-iv.rs
impl Solution {
    pub fn combination_sum4(mut nums: Vec<i32>, target: i32) -> i32 {
        nums.sort();
        let mut buff: [i32; 1001] = [0; 1001];
        buff[0] = 1;
        for i in 1..=target as usize {
            for j in &nums {
                let j = *j as usize;
                if i < j {
                    break;
                }
                buff[i] += buff[i - j];
            }
        }
        buff[target as usize]
    }
}